website/blog/2023-02-10.ferretdb-compare-arrays.md

```diff
@@ -0,0 +1,605 @@
+  { _id: 'nested-mixed', v: [ [ 'a', 1 ] ] },    // [ 'a', 1 ] is array and greater comparison order than 'c'.
+  { _id: 'nested-string', v: [ [ 'g', 'a' ] ] }, // [ 'g', 'a' ] is array and greater comparison order than 'c'.
+  { _id: 'nested-number', v: [ [ 10, 1 ] ] },    // [ 10, 1 ] is array and greater comparison order than 'c'.
+  { _id: 'array-bool', v: [ true ] }             // true is greater in comparison order than 'c'.
+]
+```
+
+Using array filter `['c']` on `$gt` operator,
+it selected `array-bool` which has boolean BSON type `true`
+that is greater than string `'c'`. it selected.
+It also selected all nested arrays 
+`nested-empty`, `nested-mixed`, `nested-string` and `nested-number` from the collection.
+The results `nested-empty`, `nested-mixed` and `nested-number` were 
+unexpected since the empty array particularly are not greater than other arrays.
+The potential explanation is that `$gt` on nested array works on BSON comparison order
+when filter is an array. However, this is different behaviour from `$lt`, which seems to compare
+the smallest element of an array.
+
+## Comparison in Update Operator
+
+Update operators `$max` and `$min` update the records that
+satisfy the comparison condition.
+
+Updating with `$max` and `$min` operators use following rules
+to compare array and other BSON types.
+
+* For array and other BSON type comparison, BSON comparison order is used.
+* Empty array is greater than null using BSON comparison order.
+
+Let's use following example to update arrays using `$max` update operator.
+The field `op` is used for the update operator to update multiple documents at once.
+
+```js
+db.updatemixed.insertMany([
+  { _id: 'bool', v: true, op: true },
+  { _id: 'array-string', v: [ 'b', 'h' ], op: true },
+  { _id: 'array-mixed', v: [ 'g', 1 ], op: true },
+  { _id: 'number', v: 4, op: true }
+])
+```
+
+`$max` operator updates the document if the specified value is greater than
+the value exists in the document. In this example `{$max: {v: 'c'}}` is used
+and this updates document if specified value `'c'` is greater than the existing
+value in the field `v` of documents.
+
+```js
+db.updatemixed.updateMany({op:true}, {$max: {v: 'c'}})
+{
+  acknowledged: true,
+  insertedId: null,
+  matchedCount: 4,
+  modifiedCount: 1,
+  upsertedCount: 0
+}
+db.updatemixed.find()
+[
+  { _id: 'bool', v: true, op: true },
+  { _id: 'array-string', v: [ 'b', 'h' ], op: true },
+  { _id: 'array-mixed', v: [ 'g', 1 ], op: true },
+  { _id: 'number', v: 'c', op: true }
+]
+```
+
+Only `number` was updated, because BSON comparison
+order of string `'c'` is greater than number `4`.
+
+No array was updated because array has greater BSON comparison order than string.
+For `$max` operator, it does not use element of an array for comparison unlike sorting
+and comparison filter operator.
+
+### Update Operator $max with Array
```

Comment:
```suggestion
### Update Operator `$max` with Array
```

maybe in such examples we can also format operators as code for consistency with other places (but it's not critical)

Reply:
Let me go through them again :pray: